src/rt/sched/behaviourcore.h

```diff
@@ -385,73 +385,95 @@ namespace verona::rt
         {
           std::get<0>(indexes[idx]) = i;
           std::get<1>(indexes[idx]) = &slots[j];
+          std::get<2>(indexes[idx]) = idx;
           idx++;
         }
       }
       auto compare = [](
-                       const std::tuple<size_t, Slot*> i,
-                       const std::tuple<size_t, Slot*> j) {
+                       const std::tuple<size_t, Slot*, size_t> i,
+                       const std::tuple<size_t, Slot*, size_t> j) {
 #ifdef USE_SYSTEMATIC_TESTING
-        return std::get<1>(i)->cown->id() > std::get<1>(j)->cown->id();
+        return std::get<1>(i)->cown->id() == std::get<1>(j)->cown->id() ?
+          std::get<2>(i) < std::get<2>(j) :
+          std::get<1>(i)->cown->id() < std::get<1>(j)->cown->id();
 #else
-        return std::get<1>(i)->cown > std::get<1>(j)->cown;
+        return std::get<1>(i)->cown == std::get<1>(j)->cown ?
+          std::get<2>(i) < std::get<2>(j) :
+          std::get<1>(i)->cown < std::get<1>(j)->cown;
 #endif
       };
 
       if (count > 1)
         std::sort(indexes.get(), indexes.get() + count, compare);
 
       // First phase - Acquire phase.
-      Cown *prev_cown = nullptr;
-      for (size_t i = 0; i < count; i++)
+      size_t i = 0;
+      while (i < count)
       {
         auto cown = std::get<1>(indexes[i])->cown;
         auto body = bodies[std::get<0>(indexes[i])];
-        auto prev = cown->last_slot.exchange(
-          std::get<1>(indexes[i]), std::memory_order_acq_rel);
-        auto* ec_ptr = &ec[std::get<0>(indexes[i])];
+        auto last_slot = std::get<1>(indexes[i]);
+        auto first_body = body;
+        size_t ii = i;
```

Comment:
`ii` unused?

Reply:
It is used to increase the ec count of the first slot in case prev was nullptr. I should give a more meaningful name.